libs/math/tests/unit/tensor/basic_tests.cpp

```diff
@@ -38,15 +38,15 @@ Tensor<fetch::fixed_point::FixedPoint<I, F>,
        fetch::memory::SharedArray<fetch::fixed_point::FixedPoint<I, F>>>
 RandomArray(std::size_t n, fetch::fixed_point::FixedPoint<I, F> adj)
 {
-  static fetch::random::LinearCongruentialGenerator gen;
+  static fetch::random::Random gen;
   Tensor<fetch::fixed_point::FixedPoint<I, F>,
          fetch::memory::SharedArray<fetch::fixed_point::FixedPoint<I, F>>>
       a1(n);
 
   fetch::fixed_point::FixedPoint<I, F> rn{0};
   for (std::size_t i = 0; i < n; ++i)
   {
-    rn       = fetch::fixed_point::FixedPoint<I, F>(gen.AsDouble());
+    rn       = fetch::fixed_point::FixedPoint<I, F>(gen.generator.AsFP64());
```

Comment:
in the future we should probably rework this 'generator' since it uses the mersenne twister from the std library - in other words, it's a source of indeterminacy - however since it's only in a test and doesn't seem to ever cause failures it's not a priority right now

Reply:
It's the fetch RNG, which is the lagged fibonacci generator - I don't think it uses the std library?